### PR TITLE
feature: ARSN-128 moved storage and algos from 8

### PR DESCRIPTION
**Pull request by miniscruff**
Will squash commits after reviews

### CODE REVIEW

File: lib/algos/list/delimiterVersions.js [hunk 1]
```diff
@@ -190,11 +190,15 @@ class DelimiterVersions extends Delimiter {
      *  @return {number}          - indicates if iteration should continue
      */
     filterV1(obj) {
+        if (Version.isPHD(obj.value)) {
+            // return accept to avoid skipping the next values in range
+            return FILTER_ACCEPT;
+        }
```
Comment (vrancurel) [inline lib/algos/list/delimiterVersions.js:196]: @jonathan-gramain is it OK ?

File: lib/algos/list/skip.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+const { FILTER_END, FILTER_SKIP, SKIP_NONE } = require('./tools');
+
+
+const MAX_STREAK_LENGTH = 100;
+
+/**
+ * Handle the filtering and the skip mechanism of a listing result.
+ */
+class Skip {
```
Comment (vrancurel) [inline lib/algos/list/skip.js:11]: @jonathan-gramain Metadata could use this class.

**Review by vrancurel — COMMENTED** [review]
Please move backend dependencies (bucketclient, hdclient, sproxydclient, ...) in the package.json

File: lib/storage/metadata/MetadataWrapper.js [hunk 3]
```diff
@@ -54,13 +55,14 @@ function _parseListEntries(entries) {
...
-function parseListEntries(entries) {
+function parseListEntries(entries, parser) {
     // wrap private function in a try/catch clause
     // just in case JSON parsing throws an exception
     try {
-        return _parseListEntries(entries);
+        return parser(entries);
     } catch (e) {
         return e;
     }
```
Comment (miniscruff) [inline lib/storage/metadata/MetadataWrapper.js:68]: Metadata wrapper still has bucketclient for now. This is due to bucketclient importing Arsenal and creating a circular dependency yarn can not resolve. This is easily fixed in the move to Armory.

**Review by alexanderchan-scality — APPROVED** [review]

**Review by vrancurel — APPROVED** [review]

**Pull request merged**